File: leetcode/valid-palindrome.cpp
#include <bits/stdc++.h>
#include <unordered_map>
#include <string>
#include <cctype>
#include <vector>
#include <algorithm>
#include <iostream>
#include <variant>
#include <any>
#include <ctype.h>
#include <stdio.h>
using namespace std;
class Solution {
public:
bool isPalindrome(string s) {
    string lit;
    lit = s;
    vector<char> newlit;

    for (char x : lit) {
        if (isalnum(static_cast<unsigned char>(x))) {
            if (isalpha(static_cast<unsigned char>(x))) {
                newlit.push_back(tolower(x));
            }

            else {newlit.push_back(x);}
        }
    }

    for (int i = 0; i < (newlit.size())/2; ++i) {
        if (newlit[i] != newlit[newlit.size()-1-i]) {
            return false;
        }
    }
    return true;
}
};

File: leetcode/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        d = []
        for x in s:
            if x.isalnum():
                d.append(x.lower())

        
        list_test = []
        list_test2 = []
        if len(d) % 2 == 0:
            for x in range(len(d) // 2):
                list_test.append(d[x])
            for x in range(len(d) - 1, len(d) // 2 - 1, -1):
                list_test2.append(d[x])
        else:
            for x in range((len(d) - 1) // 2):
                list_test.append(d[x])
            for x in range(len(d) - 1, ((len(d) - 1) // 2), -1):
                list_test2.append(d[x])
        if list_test2 == list_test:
            return True
        else:
            return False